### ISSUE TITLE
amanice.ru - authorization through social networks is blocked

### ISSUE THREAD

**Issue by Alex-302** [AdguardTeam/PopupBlocker#94]
![image](https://user-images.githubusercontent.com/8361299/37654254-23d8e4be-2c4a-11e8-9b89-4b768bee1e1c.png)

**Comment by seanl-adg** [comment]
This is caused by a flaw of overlay detection logic. Currently, when retrieving `Event` object representing user action, the Popup Blocker may retrieve it from ancestor frames, because websites may call `open` from descendant frames. From the event, we retrieve its target and check whether it looks like an artificial overlay. However, I made a mistake that the size of target element is only compared with the 'current' frame, where the correct way would be to compare it with the frame from where we retrieved the event.

In order to fix this, we should use `window` object from [`UIEvent.view` property](https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/view). Also we need to take account of this in future refactoring to get rid of `eval`.
